airflow/hooks/presto_hook.py

```diff
@@ -134,5 +134,17 @@ def insert_rows(self, table, rows, target_fields=None):
         :type rows: iterable of tuples
         :param target_fields: The names of the columns to fill in the table
         :type target_fields: iterable of strings
+        :param commit_every: The maximum number of rows to insert in one
+            transaction. Set to 0 to insert all rows in one transaction.
+        :type commit_every: int
         """
-        super().insert_rows(table, rows, target_fields, 0)
+        with closing(self.get_conn()) as conn:
+            if conn.isolation_level == 0:
```

Comment:
```suggestion
            if conn.isolation_level == IsolationLevel.AUTOCOMMIT:
```
right?

Reply:
That said it seems expensive to get a connection just to check this setting -- is there some way of doing this the doesn't invovle opening a connection, checking this setting, throwing it away and then opening it again in the `super()` call?

Reply:
One way could be checking this directly from the airflow connection itself, instead of creating a new prestosql connection.